src/components/AliasForm.vue

```diff
@@ -92,89 +129,165 @@ export default {
 			type: Object,
 			required: true,
 		},
+		enableUpdate: {
+			type: Boolean,
+			default: true,
+		},
+		enableDelete: {
+			type: Boolean,
+			default: true,
+		},
+		onUpdateSmimeCertificate: {
+			type: Function,
+			default: async (aliasId, smimeCertificateId) => {},
+		},
+		onUpdateAlias: {
+			type: Function,
+			default: async (aliasId, { alias, name }) => {},
+		},
+		onDelete: {
+			type: Function,
+			default: async (aliasId) => {},
+		},
 	},
 	data() {
 		return {
 			changeAlias: this.alias.alias,
 			changeName: this.alias.name,
+			changeSmimeCert: undefined,
 			showForm: false,
+			showSmimeForm: false,
 			loading: false,
 		}
 	},
-	methods: {
-		async updateAlias(e) {
-			this.loading = true
+	computed: {
+		...mapGetters({
+			smimeCertificates: 'getSmimeCertificates',
+		}),
+		formId() {
+			return `alias-form-${this.alias.id}`
+		},
+		smimeCertOptions() {
+			// Only show certificates that are at least valid until tomorrow
+			const now = (new Date().getTime() / 1000) + 3600 * 24
 
-			await this.$store.dispatch('updateAlias', {
-				account: this.account,
-				aliasId: this.alias.id,
-				alias: this.changeAlias,
-				name: this.changeName,
-			})
+			return this.smimeCertificates
```

Comment:
I have the feeling this always return an empty array with the self-signed cert. At least I'm not able to test currently. There is no cert selector in my alias settings.

Reply:
Same for me :confused: 

```
openssl_x509_checkpurpose($certificate, X509_PURPOSE_SMIME_SIGN, $caBundle),
openssl_x509_checkpurpose($certificate, X509_PURPOSE_SMIME_ENCRYPT, $caBundle),
```

SIGN and ENCRYPT are both false.

- Used gen-smime-cert.sh to generate CA and certificates.
- Did not import the CA. 
- Uploaded both certificate to mail.

Reply:
I'm able to select the certificate when importing the CA via occ.

Signing a message also works :partying_face: 

The signature is shown in Thunderbird as untrusted. After importing the CA it's shown as valid.

Reply:
Interesting. So the checkpurposes method also verifies the local certificate chain.

I updated the instructions above.